uber/utils.py

```diff
@@ -70,7 +76,7 @@ def check_csrf(csrf_token):
     assert csrf_token, 'CSRF token missing'
     if csrf_token != cherrypy.session['csrf_token']:
         log.error("csrf tokens don't match: {!r} != {!r}", csrf_token, cherrypy.session['csrf_token'])
-        raise AssertionError('CSRF check failed')
+        raise CSRFException('CSRF check failed')
     else:
         cherrypy.request.headers['CSRF-Token'] = csrf_token
```

Comment:
here's what I'd suggest.  replace the ```assert csrf_token``` with a CSRFException.  then in client code, only catch CSRFExceptions, log the error, then don't re-raise